### ISSUE TITLE
Add method remove_if to builder with signature build -> bool

### ISSUE THREAD

**Issue by solvingj** [conan-io/conan-package-tools#249]
The README shows the following example for filtering builds:
```python
    filtered_builds = []
    for settings, options, env_vars, build_requires, reference in builder.items:
        if settings["compiler.version"] != "4.6" and settings["build_type"] != "Debug":
             filtered_builds.append([settings, options, env_vars, build_requires])
    builder.builds = filtered_builds
```

IMO, for short ones, it would be more readable to be able to write something like the following: 

```python
builder.remove_if(lambda build: build.setting["compiler.version"] != "4.6")
```

For slightly longer ones, would like to write something like:
```python
def is_gcc_46_debug(build):
    return build.setting["compiler.version"] != "4.6"` and build.settings["build_type"] != "Debug"

builder.remove_if(lambda build: is_gcc_46_debug(build))
```

I think it's not too hard to implement.  I don't know if we can even do `build.setting` or `build["setting"]`, nor do i know anything about named builds.  I was going to try to implement myself quickly until I saw that, so it will take some extra time to see how that works.  Please just let me know if there are any significant barriers to this one.

**Comment by lasote** [comment]
I think this is a good idea!

**Comment by lasote** [comment]
To be released in 0.20.0
